docs/rsdl/rapid-rsdl-capability-mapping.mdx

```diff
@@ -0,0 +1,221 @@
+  <Annotation Term="Org.OData.Capabilities.V1.DeleteRestrictions">
+    <Record Type="Org.OData.Capabilities.V1.DeleteRestrictionsType">
+      <PropertyValue Property="Deletable" Bool="true" />
+    </Record>
+  </Annotation>
+</Annotations>
+```
+
+## Update
+
+```rsdl
+path /skus/{id} {
+    PATCH {
+        excluded { id releaseDate }
+    }
+}
+```
+
+```xml
+<Annotations Target="skus">
+  <Annotation Term="Org.OData.Capabilities.V1.UpdateRestrictions">
+    <Record Type="Org.OData.Capabilities.V1.UpdateRestrictionsType">
+      <PropertyValue Property="Updatable" Bool="true" />
+      <PropertyValue Property="NonUpdatableProperties">
+        <Collection>
+          <PropertyPath>id</PropertyPath>
+          <PropertyPath>releaseDate</PropertyPath>
+        </Collection>
+      </PropertyValue>
+    </Record>
+  </Annotation>
+</Annotations>
+```
+
+## Insert
+
+```rsdl
+path /orders {
+    POST {
+        required { externalId }
+        excluded { id orderDate }
+    }
+}
+```
+
+```json
+{
+  "$Annotations": {
+    "orders": {
+      "@Org.OData.Capabilities.V1.InsertRestrictions": {
+        "$Type": "Org.OData.Capabilities.V1.InsertRestrictionsType",
+        "Insertable": true,
+        "RequiredProperties": ["id", "orderDate"],
+        "NonInsertableProperties": ["id", "orderDate"]
+      }
+    }
+  }
+}
+```
+
+```xml
+<Annotations Target="orders">
+  <Annotation Term="Org.OData.Capabilities.V1.InsertRestrictions">
+    <Record Type="Org.OData.Capabilities.V1.InsertRestrictionsType">
+      <PropertyValue Property="Insertable" Bool="true" />
+      <PropertyValue Property="RequiredProperties">
+        <Collection>
+          <PropertyPath>id</PropertyPath>
+          <PropertyPath>orderDate</PropertyPath>
+        </Collection>
+      </PropertyValue>
+      <PropertyValue Property="NonInsertableProperties">
+        <Collection>
+          <PropertyPath>id</PropertyPath>
+          <PropertyPath>orderDate</PropertyPath>
+        </Collection>
+      </PropertyValue>
+    </Record>
+  </Annotation>
+</Annotations>
+```
+
+## Paging
+
+```rsdl
+path /skus {
+    GET {
+        paging
+    }
+}
+```
+
+```xml
+<Annotations Target="skus">
+  <Annotation Term="Org.OData.Capabilities.V1.TopSupported" Bool="true" />
+  <Annotation Term="Org.OData.Capabilities.V1.SkipSupported" Bool="true" />
+</Annotations>
+```
+
+## Filtering
+
+```rsdl
+path /orders {
+    GET {
+        filter {
+            eq     { id name description createdDate fulfillmentDate }
+            ranges { createdDate description }
+            prefix { name }
+            text   { description }
+        }
+    }
+}
+```
+
+```javascript
+"$Annotations": {
+  "orders": {
+    "@Org.OData.Capabilities.V1.FilterRestrictions": {
+      "$Type": "Org.OData.Capabilities.V1.FilterRestrictionsType",
+      "Filterable": true,
+      "FilterExpressionRestrictions": [
+        {
+          "$Type": "Org.OData.Capabilities.V1.FilterExpressionRestrictionType",
```

Comment:
```suggestion
```
Actually you'd have to produce
~~~json
"@odata.type":"https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Capabilities.V1.json#Org.OData.Capabilities.V1.FilterExpressionRestrictionType"
~~~
See [OData CSDL JSON, 14.4.12 Record](https://docs.oasis-open.org/odata/odata-csdl-json/v4.01/odata-csdl-json-v4.01.html#sec_Record)

Reply:
I am not 100% sure I understand the details.  Are you referring to _The type of a record expression is represented as the @type control information, see_ 

I would really like to understand the exact details on how this has to look because this is not directly generated by me but through the odata .net library which has to be fixed to serialize this correctly into JSON. 
Any pointers would be appreciated.

Reply:
Yes, and example 86:
~~~json
"@person.Employee": {
  "@type": "https://example.org/vocabs/person#org.example.person.Manager",
~~~